Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -122,14 +124,69 @@ void CookieStore::getAll(CookieStoreGetOptions&&, Ref<DeferredPromise>&& promise
     promise->reject(NotSupportedError);
 }
 
-void CookieStore::set(const String&, const String&, Ref<DeferredPromise>&& promise)
+void CookieStore::set(String&& name, String&& value, Ref<DeferredPromise>&& promise)
 {
-    promise->reject(NotSupportedError);
+    set(CookieInit { WTFMove(name), WTFMove(value) }, WTFMove(promise));
 }
 
-void CookieStore::set(CookieInit&&, Ref<DeferredPromise>&& promise)
+void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
 {
-    promise->reject(NotSupportedError);
+    auto* context = scriptExecutionContext();
+    if (!context) {
+        promise->reject(SecurityError);
+        return;
+    }
+
+    auto* origin = context->securityOrigin();
+    if (!origin) {
+        promise->reject(SecurityError);
+        return;
+    }
+
+    if (origin->isOpaque()) {
+        promise->reject(Exception { SecurityError, "The origin is opaque"_s });
+        return;
+    }
+
+    auto& document = *downcast<Document>(context);
```

Comment:
is this safe to do?

Reply:
Yes, it was null-checked earlier and the constructor takes a Document as parameter (which it passes to ActiveDOMObject, and which is what ActiveDOMObject::scriptExecutionContext() returns)